rule/time_date.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	w := &lintTimeDate{file, onFailure}
+
+	ast.Walk(w, file.AST)
+	return failures
+}
+
+// Name returns the rule name.
+func (*TimeDateRule) Name() string {
+	return "time-date"
+}
+
+type lintTimeDate struct {
+	file      *lint.File
+	onFailure func(lint.Failure)
+}
+
+var (
+	// timeDateArgumentNames are the names of the arguments of time.Date
+	timeDateArgumentNames = []string{
+		"year",
+		"month",
+		"day",
+		"hour",
+		"minute",
+		"second",
+		"nanosecond",
+		"timezone",
+	}
+
+	// timeDateArity is the number of arguments of time.Date
+	timeDateArity = len(timeDateArgumentNames)
+)
+
+func (w lintTimeDate) Visit(n ast.Node) ast.Visitor {
+	ce, ok := n.(*ast.CallExpr)
+	if !ok || len(ce.Args) != timeDateArity {
+		return w
+	}
+	if !isPkgDot(ce.Fun, "time", "Date") {
+		return w
+	}
+
+	// The last argument is a timezone, the is no need to check it, also it has a different type
+	for pos, arg := range ce.Args[:timeDateArity-1] {
+		bl, ok := arg.(*ast.BasicLit)
+		if !ok {
+			continue
+		}
+
+		replacedValue, err := parseDecimalInteger(bl)
+		if err == nil {
+			continue
+		}
+
+		if errors.Is(err, errParsedInvalid) {
+			// this is not supposed to happen
+			// let's be defensive and skip this one
+
+			logger, errLogger := logging.GetLogger()
+			if errLogger != nil {
+				// this is not supposed to happen
+				// we discard the error
+				continue
+			}
+			logger.With(
+				"value", bl.Value,
+				"kind", bl.Kind,
+				"error", err.Error(),
+			).Error("failed to parse time.Date argument")
+
+			continue
+		}
+
+		confidence := 0.8 // default confidence
+		errMessage := err.Error()
+		instructions := fmt.Sprintf("use %s instead of %s", replacedValue, bl.Value)
+		if errors.Is(err, errParsedOctal) {
+			switch {
+			case errors.Is(err, errParsedOctalWithZero):
+				// people are allowed to use 00, 01, 02, 03, 04, 05, 06, and 07
+				confidence = 0.5
+
+			case errors.Is(err, errParsedOctalWithPaddingZeroes):
+				confidence = 1 // this is a clear mistake
+				// example with 000123456 (octal) is about 123456 or 42798 ?
+
+				strippedValue := strings.TrimLeft(bl.Value, "0")
+				if strippedValue == "" {
+					// avoid issue with 00000000
+					strippedValue = "0"
+				}
+
+				if strippedValue != replacedValue {
+					instructions = fmt.Sprintf(
+						"choose between %s and %s (decimal value of %s octal value)",
+						strippedValue, replacedValue, strippedValue,
+					)
+				}
+			}
+		}
+
+		w.onFailure(lint.Failure{
+			Category:   "time",
+			Node:       bl,
+			Confidence: confidence,
+			Failure: fmt.Sprintf(
+				"use decimal digits for time.Date %s argument: %s found: %s",
+				timeDateArgumentNames[pos], errMessage, instructions),
+		})
+	}
+
+	return w
+}
+
+var (
+	errParsedOctal                  = errors.New("octal notation")
+	errParsedOctalWithZero          = fmt.Errorf("%w with leading zero", errParsedOctal)
+	errParsedOctalWithPaddingZeroes = fmt.Errorf("%w with padding zeroes", errParsedOctal)
```

Comment:
Why not?
```suggestion
	errParsedOctalWithZero          = errors.New("octal notation with leading zero")
	errParsedOctalWithPaddingZeroes = errors.New("octal notation with padding zeroes")
```
It's much simpler and nicer.

Reply:
Because of this

https://github.com/mgechev/revive/pull/1327/files#diff-e4fddf806b4471662eca22fd05b3a8d7feb2a1fde48a18eed97620d7b419b4adR108